backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -1446,4 +1450,127 @@ private boolean checkAllResourceAllowed(List<Resources> resources) {
             this::getEntityPermissionBasedOnResourceTypes)
         .thenApply(versioningEntryMap -> versioningEntryMap.get(request.getId()), executor);
   }
+
+  public InternalFuture<CloneExperimentRun.Response> cloneExperimentRun(
+      CloneExperimentRun request) {
+    var validateRequestParamFuture =
+        InternalFuture.runAsync(
+            () -> {
+              if (request.getSrcExperimentRunId().isEmpty()) {
+                throw new InvalidArgumentException("Source ExperimentRun Id should not be empty");
+              }
+            },
+            executor);
+
+    return validateRequestParamFuture
+        .thenCompose(
+            unused ->
+                FutureGrpc.ClientRequest(
+                    uac.getUACService().getCurrentUser(Empty.newBuilder().build()), executor),
+            executor)
+        .thenCompose(
+            userInfo ->
+                findExperimentRuns(
+                        FindExperimentRuns.newBuilder()
+                            .addExperimentRunIds(request.getSrcExperimentRunId())
+                            .build())
+                    .thenApply(
+                        findExperimentRuns -> {
+                          if (findExperimentRuns.getExperimentRunsList().isEmpty()) {
+                            throw new PermissionDeniedException("Permission Denied");
+                          }
```

Comment:
This might be permission denied or not found, right?

Reply:
Changed to NOT_FOUND